### ISSUE TITLE
Broken ngStep for input[type="number"] directive

### ISSUE THREAD

**Issue by lucassus** [angular/angular.js#15504]
**Do you want to request a *feature* or report a *bug*?**

It's a *bug*

**What is the current behavior?**

For values like `1.16`, `20.1` etc the `input[type=number]` field with `ng-step="0.01"` is marked as invalid.

Steps to reproduce:

1. Update to AngularJS version `1.6.0`
2. Define an `input[type=number]` with `ng-step="0.01"`
3. For values like `1.16`, `20.1` the field will be marked as `$invalid`

See a small demo: https://plnkr.co/edit/LW2BsaA7FfEVu9nB6RoG?p=preview

I believe it's a problem with floating points arithmetic and this little method https://github.com/angular/angular.js/blob/master/src/ng/directive/input.js#L1563

```javascript
    // ...

    value = value * multiplier;
    stepBase = stepBase * multiplier;
    step = step * multiplier;
  }

  return (value - stepBase) % step === 0;
```

When for instance `value = 1.16` and `multiplier = 100` `value * multiplier` gives us `115.99999999999999` so the return expression returns false even when the given value is valid.

**Comment by gkalpak** [comment]
BTW, this also affects `ngStep` on `input[range]` (since both input types share their `step` implementation).

**Comment by benneq** [comment]
this should also be added to 1.6.0

**Comment by gkalpak** [comment]
This has not been fixed yet. Once it is fixed, it will be included in 1.5.x and 1.6.x.
